pkg/genericapiserver/enablement_config.go

```diff
@@ -0,0 +1,159 @@
+/*
+Copyright 2016 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package genericapiserver
+
+import (
+	"k8s.io/kubernetes/pkg/api/unversioned"
+	"k8s.io/kubernetes/pkg/util/sets"
+)
+
+// APIResourceEnablement is the interface to determine which versions and resources are enabled
+type APIResourceEnablement interface {
```

Comment:
> Do you need this interface to be declared here? I'd actually like to use the name for the struct.

Huh?  This is the interface that consumers of the `genericapiserver.Config` would use.  We should not be forcing them to use an impl as they do today.  This interface is the contract used by the `genericapiserver.Config` and I'm just providing a default implementation suited to kube's current needs here.

The whole point of these pulls is that consumer (openshift) should have the option to use any mechanism we want to determine enablement and the `genericapiserver.Config` only cares about getting something that satisfies this interface.  But only this code can provide it.

Are you saying something else or are we in violent agreement?